crates/issue/src/lib.rs

```diff
@@ -372,67 +364,48 @@ impl<T: Config> Pallet<T> {
+            IssueRequestStatus::Cancelled => {
+                // if vault is not accepting new issues, we don't allow the execution of cancelled
+                // issues, since this would drop the collateralization rate unexpectedly
+                ext::vault_registry::ensure_accepting_new_issues::<T>(&issue.vault)?;
 
-            // decrease the to-be-issued tokens that will not be issued after all
-            let deficit = expected_total_amount.checked_sub(&amount_transferred)?;
-            ext::vault_registry::decrease_to_be_issued_tokens::<T>(&issue.vault, &deficit)?;
+                // first try to increase the to-be-issued tokens - if the vault does not
+                // have sufficient collateral then this aborts
+                ext::vault_registry::try_increase_to_be_issued_tokens::<T>(&issue.vault, &amount_transferred)?;
 
-            // slash/release griefing collateral proportionally to the amount sent
-            let released_collateral = ext::vault_registry::calculate_collateral::<T>(
-                &issue.griefing_collateral(),
-                &amount_transferred,
-                &expected_total_amount,
-            )?;
-            released_collateral.unlock_on(&requester)?;
-            let slashed_collateral = issue.griefing_collateral().checked_sub(&released_collateral)?;
-            ext::vault_registry::transfer_funds::<T>(
-                CurrencySource::UserGriefing(issue.requester.clone()),
-                CurrencySource::FreeBalance(issue.vault.account_id.clone()),
-                &slashed_collateral,
-            )?;
-
-            Self::update_issue_amount(&issue_id, &mut issue, amount_transferred, slashed_collateral)?;
-        } else {
-            // release griefing collateral
-            let griefing_collateral: Amount<T> = issue.griefing_collateral();
-            griefing_collateral.unlock_on(&requester)?;
-
-            if amount_transferred.gt(&expected_total_amount)?
-                && !ext::vault_registry::is_vault_liquidated::<T>(&issue.vault)?
-            {
-                let surplus_btc = amount_transferred.checked_sub(&expected_total_amount)?;
-
-                match ext::vault_registry::try_increase_to_be_issued_tokens::<T>(&issue.vault, &surplus_btc) {
-                    Ok(_) => {
-                        // Current vault can handle the surplus; update the issue request
-                        Self::update_issue_amount(
+                if amount_transferred.lt(&expected_total_amount)? {
+                    ensure!(requester == executor, Error::<T>::InvalidExecutor);
+                }
+                if amount_transferred.ne(&expected_total_amount)? {
+                    let slashed = Amount::zero(T::GetGriefingCollateralCurrencyId::get());
+                    Self::set_issue_amount(&issue_id, &mut issue, amount_transferred, slashed)?;
+                }
+            }
+            IssueRequestStatus::Pending => {
```

Comment:
Since this branch is now callable after expiry I wonder if we should anyway slash the collateral for being late?

Reply:
Hmm I don't think so. It'd only complicate the code more - I'd just leave it as the vault's responsibility to cancel asap if it wants